src/hotspot/share/opto/superword.cpp

```diff
@@ -2237,396 +1992,65 @@ void SuperWord::apply_memops_reordering_with_schedule(Node_List& memops_schedule
   }
 }
 
-// Convert packs into vector node operations
-// At this point, all correctness and profitability checks have passed.
-// We start the irreversible process of editing the C2 graph. Should
-// there be an unexpected situation (assert fails), then we can only
-// bail out of the compilation, as the graph has already been partially
-// modified. We bail out, and retry without SuperWord.
-bool SuperWord::apply_vectorization() {
-  CountedLoopNode *cl = lpt()->_head->as_CountedLoop();
-  assert(cl->is_main_loop(), "SLP should only work on main loops");
+// We call "apply" on every VTransformNode, which replaces the packed scalar nodes
+// with vector nodes.
+void VTransformGraph::apply_vectorization() const {
   Compile* C = phase()->C;
-  assert(!_packset.is_empty(), "vectorization requires non-empty packset");
-
 #ifndef PRODUCT
-  if (TraceLoopOpts) {
-    tty->print("SuperWord::apply_vectorization ");
-    lpt()->dump_head();
+  if (_is_trace_verbose) {
+    tty->print_cr("\nVTransformGraph::apply_vectorization:");
   }
 #endif
 
-  uint max_vlen_in_bytes = 0;
-  uint max_vlen = 0;
-
-  for (int i = 0; i < body().length(); i++) {
-    Node* n = body().at(i);
-    Node_List* p = get_pack(n);
-    if (p != nullptr && n == p->at(p->size()-1)) {
-      // After apply_memops_reordering_with_schedule, we know that the memops have the same order in the pack
-      // as in the memory slice. Hence, "first" is the first memop in the slice from the pack,
-      // and "n" is the last node in the slice from the pack.
-      Node* first = p->at(0);
-      uint vlen = p->size();
-      uint vlen_in_bytes = 0;
-      Node* vn = nullptr;
-      int   opc = n->Opcode();
-      if (n->is_Load()) {
-        Node* ctl = n->in(MemNode::Control);
-        Node* mem = first->in(MemNode::Memory);
-        // Set the memory dependency of the LoadVector as early as possible.
-        // Walk up the memory chain, and ignore any StoreVector that provably
-        // does not have any memory dependency.
-        while (mem->is_StoreVector()) {
-          VPointer p_store(mem->as_Mem(), _vloop);
-          if (p_store.overlap_possible_with_any_in(p)) {
-            break;
-          } else {
-            mem = mem->in(MemNode::Memory);
-          }
-        }
-        Node* adr = first->in(MemNode::Address);
-        const TypePtr* atyp = n->adr_type();
-        vn = LoadVectorNode::make(opc, ctl, mem, adr, atyp, vlen, velt_basic_type(n), control_dependency(p));
-        vlen_in_bytes = vn->as_LoadVector()->memory_size();
```

Comment:
Note:
Captured into `VTransformLoadVectorNode` by `SuperWordVTransformBuilder::make_vtnode_for_pack`. C2 nodes generated in `VTransformLoadVectorNode::apply`.